onadata/apps/api/tests/viewsets/test_dataview_viewset.py

```diff
@@ -178,16 +179,63 @@ def test_dataview_with_attachment_field(self):
         attachment_list_view = AttachmentViewSet.as_view({"get": "list"})
         request = self.factory.get("/?dataview=" + str(self.data_view.pk), **self.extra)
         response = attachment_list_view(request)
+        attachments = Attachment.objects.filter(
+            instance__xform=self.data_view.xform)
         self.assertEqual(1, len(response.data))
-        self.assertEqual(self.data_view.query, {})
+        self.assertEqual(self.data_view.query,
+                         [{'value': 'no', 'column': 'pizza_fan', 'filter': '='}])
         serialized_attachments = AttachmentSerializer(
-                Attachment.objects.filter(
-                    instance__xform=self.data_view.xform),
-                many=True, context={'request': request}).data
+            attachments,
+            many=True, context={'request': request}).data
         self.assertEqual(
             serialized_attachments,
             response.data)
 
+        # create profile for alice
+        alice_data = {'username': 'alice', 'email': 'alice@localhost.com',
+                      'password1': 'alice', 'password2': 'alice',
+                      'first_name': 'Alice', 'last_name': 'A',
+                      'city': 'Nairobi', 'country': 'KE'}
+        alice_profile = self._create_user_profile(extra_post_data=alice_data)
+
+        # check that user with no permisisons can not list attachment objects
+        request = self.factory.get("/?dataview=" + str(self.data_view.pk), **self.extra)
+        request.user = alice_profile.user
+        response = attachment_list_view(request, user=alice_profile.user)
+        attachments = Attachment.objects.filter(
+            instance__xform=self.data_view.xform)
+        self.assertEqual(0, len(response.data))
+        self.assertEqual(self.data_view.query,
+                         [{'value': 'no', 'column': 'pizza_fan', 'filter': '='}])
+        self.assertEqual(
+            [],
+            response.data)
+
+        # check that user with no permisisons can not view a specific attachment object
+        attachment_list_view = AttachmentViewSet.as_view({"get": "retrieve"})
+        request = self.factory.get("/?dataview=" + str(self.data_view.pk), **self.extra)
+        request.user = alice_profile.user
```

Comment:
Why not generate a different authentication extra for this request ? `self.extra` usually contains the authentication credentials for the test user, the request authentication class will set the correct `request.user` attribute down the line